test/metabase/query_processor/middleware/add_dimension_projections_test.clj

```diff
@@ -521,19 +534,20 @@
                                          {:base-type :type/Decimal,    :cast-fn bigdec}
                                          {:base-type :type/BigInteger, :cast-fn bigint}
                                          {:base-type :type/Text,       :cast-fn str}]]
-      (testing (format "Base type = %s; IDs in result rows are %s" base-type (class (cast-fn 1)))
-        (let [info (#'qp.add-dimension-projections/col->dim-map
-                    1
-                    (assoc (meta/field-metadata :venues :category-id)
-                           :base-type          base-type
-                           :effective-type     base-type
-                           :lib/internal-remap {:lib/type              :metadata.column.remapping/internal
-                                                :id                    1
-                                                :name                  "Category ID [internal remap]"
-                                                :values                [24]
-                                                :human-readable-values ["Fashion"]}))
-              f    (#'qp.add-dimension-projections/make-row-map-fn [info])]
-          (is (= [["Some store" (cast-fn 1) nil]
-                  ["Another store" (cast-fn 24) "Fashion"]]
-                 (transduce (map f) conj [] [["Some store" (cast-fn 1)]
-                                             ["Another store" (cast-fn 24)]]))))))))
+      (qp.store/with-metadata-provider meta/metadata-provider
+        (testing (format "Base type = %s; IDs in result rows are %s" base-type (class (cast-fn 1)))
+          (let [info (#'qp.add-dimension-projections/col->dim-map
+                      1
+                      (assoc (meta/field-metadata :venues :category-id)
+                             :base-type          base-type
+                             :effective-type     base-type
+                             :lib/internal-remap {:lib/type              :metadata.column.remapping/internal
+                                                  :id                    1
```

Comment:
Does comment from schema above about nil ids for internal remaps not apply here?

Reply:
Good catch.